RELEASE/scripts/autoscend/auto_quest_level_12.ash

```diff
@@ -1290,6 +1299,223 @@ boolean L12_themtharHills()
 	return true;
 }
 
+boolean L12_farm()
+{
+	if(get_property("auto_skipL12Farm").to_boolean())
+	{
+		return false;
+	}
+	if(get_property("sidequestFarmCompleted") != "none")
+	{
+		set_property("auto_skipL12Farm", "true");
+		return false;
+	}
+	if(internalQuestStatus("questL12War") != 1)
+	{
+		return false;
+	}
+	
+	string faction = "fratboy";
+	if(get_property("auto_hippyInstead").to_boolean())
+	{
+		faction = "hippy";
+	}
+	
+	if(faction == "fratboy" && get_property("hippiesDefeated").to_int() < 458)
+	{
+		return false;
+	}
+	
+	//calculate if farm is worth doing.
```

Comment:
This should be broken up into some smaller utility functions. Stuff like "how many quests have I done", "how many enemies are left on the battlefield," "how long do I expect the farm to take," "how many enemies am I killing per adventure," and maybe even a utility function that calculates the number of adventures saved by doing an additional battlefield quest?

Really good code is split up so each part is doing a super simple thing and is easy to verify.

Reply:
Fair enough.
First step taken with 3f8676e

Reply:
how long i expect quest to take and how many adventures i will be saving has different calculations for each sidequests. So it is not easy to make them into independent functions. I will try to come up with a good way to do so.

Reply:
I was imagining comparing `int auto_estimateTurnCost(war quest)` and `int auto_turnsSavedByFinishingQuest(war quest)". But there's some awkwardness here, where doing the first quest might save X adventures if we were planning on already doing quests 2, 3, 4, but save a different number of adventures if we were planning on doing quests 3, 4, 5. Hm. I will play around and try to get something working.

Reply:
commit ef25a09